src/System.Console/src/System/ConsolePal.Unix.cs

```diff
@@ -73,6 +73,36 @@ public static void ResetColor()
             }
         }
 
+        public static int WindowWidth
+        {
+            get
+            {
+                if (!ConsoleOutIsTerminal) return 0; // Should we rather throw exception here? On windows we get IOException from the native code, do we need to match for code portability?
+                return TerminalBasicInfo.Instance.ColumnFormat;
```

Comment:
By native shim do you mean the System.Native? Also, I was thinking we could try to read the environment variable COLUMNS, which does get updated on resizing the terminal. I have not yet tested it by using Environment.GetEnvironmentVariable though.